arwen/host/execution.go

```diff
@@ -170,10 +173,38 @@ func (host *vmHost) ExecuteOnDestContext(input *vmcommon.ContractCallInput) (vmO
 	return
 }
 
-func (host *vmHost) finishExecuteOnDestContext(executeErr error) *vmcommon.VMOutput {
+func computeGasUsedByCurrentSC(
+	gasUsed uint64,
+	vmOutput *vmcommon.VMOutput,
+	executeErr error,
+) (uint64, error) {
+	if executeErr != nil {
+		return 0, executeErr
+	}
+
+	if vmOutput.ReturnCode != vmcommon.Ok {
+		return 0, nil
+	}
+
+	for _, outAcc := range vmOutput.OutputAccounts {
+		if gasUsed < outAcc.GasUsed+outAcc.GasLimit {
```

Comment:
Perhaps describe (as a comment in code) when can this happen?

By the way, what happens when calling a library contract multiple times from the same transaction? Do we need some tests regarding this?

Reply:
NO - this should not happen ever - coding error.